### ISSUE TITLE
Add support for `IngressClass` objects

### ISSUE THREAD

**Issue by levikobi** [kubernetes-sigs/ingress2gateway#117]
**What would you like to be added**:
Add support for reading and taking into account `IngressClass` objects.

**Why this is needed**:
If users do not specify `ingressClassName,` we need to understand the default `IngressClass` that should be used for the conversion.

**Comment by soravkumarsharma** [comment]
I want to work on this issue

**Comment by Varun1300211** [comment]
I have created a PR for the same. Kindly review it.

**Comment by levikobi** [comment]
I just now noticed that @akagami-harsh asked to work on this issue first.
@akagami-harsh are you still interested in working on it?

**Comment by akagami-harsh** [comment]
@levikobi yes i am interested. i was busy with other things, but now i am free

**Comment by akagami-harsh** [comment]
Since there's already an open PR for this issue, should I create a new one?

**Comment by LiorLieberman** [comment]
Hi @akagami-harsh, no. 
There is already a PR @Varun1300211 opened. 

Please check out other issues like https://github.com/kubernetes-sigs/ingress2gateway/issues/81 if you are still up for contributing. We would love to have you.

**Comment by Varun1300211** [comment]
Apologies @akagami-harsh, I am still very new to the open source. Never seen you were already assigned to it before. Hope you understand.

Sure @LiorLieberman will keep it in mind from the next time onwards.

**Comment by akagami-harsh** [comment]
No worries at all, @Varun1300211! It happens, and I appreciate you being open to the feedback. Thanks for letting me know, @LiorLieberman  . I'll take a look at issue #81.

**Comment by ipsum-0320** [comment]
hi @akagami-harsh, are you still working on it, I would like to give it a try. 😊  /assign

**Comment by mlavacca** [comment]
@ipsum-0320, please notice @Varun1300211 is already assigned to this issue.

@Varun1300211, are you still working on it?

**Comment by ipsum-0320** [comment]
> @ipsum-0320, please notice @Varun1300211 is already assigned to this issue.
> 
> @Varun1300211, are you still working on it?

okok, I will wait until @Varun1300211  replies before taking the next step 😊

**Comment by VasuDevrani** [comment]
Hey @LiorLieberman @levikobi, does this approach sounds good?

1.Update `GetIngressClass` function in `pkg/i2gw/providers/common/utils.go` to consider IngressClass objects
2. Update `ReadIngressesFromCluster` function in `pkg/i2gw/providers/common/resource_reader.go` to also read IngressClass object

**Comment by VasuDevrani** [comment]
If neither the `ingressClassName` field nor the `kubernetes.io/ingress.class` annotation is set on the Ingress, the tool skips processing that Ingress, resulting in no Gateway API resources?

**Comment by levikobi** [comment]
> @levikobi @LiorLieberman If neither the `ingressClassName` field nor the `kubernetes.io/ingress.class` annotation is set on the Ingress, the tool skips processing that Ingress, resulting in no Gateway API resources?

At the time I opened this issue this was correct @VasuDevrani, I think this still holds true though.
The purpose of this issue is to add support for reading the `ingressClasses`, one of them should be the default. Then we will be able to convert ingresses without the field or annotation being specified

**Comment by mlavacca** [comment]
Hi @VasuDevrani, any news on this issue? If you need some help, please let us know :)

**Comment by TusharMohapatra07** [comment]
hey @mlavacca is this issue still open to contributions ??

**Comment by manju956** [comment]
Hi @mlavacca is the issue still relavant for me to pick up and implement?
